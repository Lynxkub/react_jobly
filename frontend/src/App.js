import React , {useEffect, useState} from 'react';
import './App.css';
import NavBar from './NavBar';
import Routes from './Routes';
import JoblyApi from './api';



function App() {
 
  const [userToken , setUserToken] = useState(localStorage.getItem('token'))
  const [username , setUsername] = useState(localStorage.getItem('username'))

  useEffect(() => {
    localStorage.setItem('token' , userToken)
  } , [userToken])


  JoblyApi.token = localStorage.getItem('token');
  
  const jobsAppliedTo = localStorage.getItem('jobs');
  if(!jobsAppliedTo){
    localStorage.setItem('jobs' , JSON.stringify([' ']));
  }
 

  const signUp = async (newUser) => {
  setUserToken(await JoblyApi.register(newUser));
  localStorage.setItem('token' , userToken)
  setUsername(localStorage.getItem('username'));
}

  const login = async (user) => {
    setUserToken(await JoblyApi.login(user))
    localStorage.setItem('token' , userToken)
    setUsername(localStorage.getItem('username'));
  }

  const logout =  () => {
    setUserToken(undefined);
    setUsername(localStorage.getItem('username'));
  

  }



  return (
    <div className='fill-window'>
    <NavBar token = {userToken} username = {username} />
    <Routes signUp = {signUp} login = {login} logout = {logout} token = {userToken} />
    </div>
  );
}

export default App;

