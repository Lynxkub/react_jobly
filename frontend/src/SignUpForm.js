import React , {useState} from 'react';
import { useHistory } from 'react-router-dom';
import './home.css'

const SignUpForm = ( { signUp , token}) => {
    const history = useHistory();
    const INITIAL_STATE = {
        username : '',
        password : '',
        firstName : '',
        lastName : '',
        email : ''
    }
    const [formData , setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const {name , value} = e.target;
        setFormData(formData => ({
            ...formData,
            [name] : value
        }))
    }


    const handleSubmit = async (e) => {
        try{
        e.preventDefault();
        localStorage.setItem('username' , formData.username)
        await signUp(formData);
        console.log(token)
        setFormData(INITIAL_STATE)
        history.push('/')
        }catch (err){
            alert(err)
        }
    }

    return(
        <div className='d-flex align-items-center bg-landing'>
            <div className='container-md bg-background w-50 p-4'>
            <form onSubmit = {handleSubmit}> 
                <label htmlFor = 'username' className='form-label text-light'>Username</label>
                <input
                type='text'
                placeholder='Username'
                name='username'
                data-testid='username'
                value={formData.username}
                onChange={handleChange} 
                className='form-control'
                />
                <label htmlFor = 'password' className='form-label text-light'>Password</label>
                <input
                type='password'
                placeholder='Password'
                name='password'
                data-testid='password'
                value={formData.password}
                onChange={handleChange}
                className='form-control' 
                />
                <label htmlFor = 'firstName' className='form-label text-light'>First Name</label>
                <input
                type='text'
                placeholder='First Name'
                name='firstName'
                data-testid='firstName'
                value={formData.firstName}
                onChange={handleChange}
                className='form-control' 
                />
                <label htmlFor = 'lastName' className='form-label text-light'>Last Name</label>
                <input
                type='text'
                placeholder='Last Name'
                name='lastName'
                data-testid='lastName'
                value={formData.lastName}
                onChange={handleChange} 
                className='form-control'
                />
                <label htmlFor = 'email' className='form-label text-light'>Email</label>
                <input
                type='email'
                placeholder='Email'
                name='email'
                data-testid='email'
                value={formData.email}
                onChange={handleChange} 
                className='form-control'
                />
                <button data-testid = 'signupButton' className='btn btn-md btn-dark'>Sign Up</button>
            </form>
            </div>
        </div>
    )
}

export default SignUpForm;