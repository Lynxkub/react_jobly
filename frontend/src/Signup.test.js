import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SignUpForm from './SignUpForm';
import JoblyApi from './api';
import App from './App';


test('it renders without crashing' , () => {
    render(
        <MemoryRouter>
            <SignUpForm />
        </MemoryRouter>
    )
})


test('it matches snapshot' , () => {
    const {asFragment} = render(
        <MemoryRouter>
            <SignUpForm />
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
})


test('sign up form successfully signs up a new user' , () => {
    const {getByText , getByTestId} = render(
        <MemoryRouter initialEntries={['/']}>
            <App />
        </MemoryRouter>
    )
    const signupLink = getByText('Sign Up')
    fireEvent.click(signupLink);

    // make sure link takes user to a sign up form
    
    const username = getByTestId('username')
    expect(username).toBeInTheDocument();

    // fill out information and submit
    const password = getByTestId('password');
    const email = getByTestId('email');
    const firstName = getByTestId('firstName');
    const lastName = getByTestId('lastName');
    const signupButton = getByTestId('signupButton')
    fireEvent.change(username , {target : {value : 'newUser'}})
    fireEvent.change(password , {target : {value : 'password'}})
    fireEvent.change(firstName , {target : {value : 'newUser'}})
    fireEvent.change(lastName , {target : {value : 'newUser'}})
    fireEvent.change(email , {target : {value : 'newUser@newUser.com'}})
    fireEvent.click(signupButton);

    setTimeout(async () => {
        const userData = await JoblyApi.getUser('newUser');
        expect(userData).toEqual({
            username: 'newUser',
            firstName : 'newUser',
            lastName : 'newUser',
            email : 'newUser@newUser.com'
        })
    } , 1000)

})