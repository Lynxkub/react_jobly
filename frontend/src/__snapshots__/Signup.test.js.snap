// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="username"
      >
        Username
      </label>
      <input
        data-testid="username"
        name="username"
        placeholder="Username"
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        data-testid="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        data-testid="firstName"
        name="firstName"
        placeholder="First Name"
        type="text"
        value=""
      />
      <label
        for="lastName"
      >
        Last Name
      </label>
      <input
        data-testid="lastName"
        name="lastName"
        placeholder="Last Name"
        type="text"
        value=""
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        data-testid="email"
        name="email"
        placeholder="Email"
        type="email"
        value=""
      />
      <button
        data-testid="signupButton"
      >
        Sign Up
      </button>
    </form>
  </div>
</DocumentFragment>
`;
